features/src/range_image_border_extractor.cpp

```diff
@@ -163,7 +162,7 @@ RangeImageBorderExtractor::extractBorderScoreImages ()
   int width  = range_image_->width,
       height = range_image_->height,
       size   = width*height;
-  border_scores_left_   = new float[size];
+  border_scores_left_.assign(size, 0);
```

Comment:
Zero-initialization is an overhead compared to the original version, right?

Reply:
That might be mitigated by using resize, since there's zero init for new members only, but no compulsory memory allocation. **It's a tradeoff**. If the size doesn't change (or decreases), it'll benefit the user. If the size increases, it'll cost more (linear in the delta increase).

Also, any kernel page (in new memory allocation) would be zero init for security reasons, which will bring further parity for large sizes